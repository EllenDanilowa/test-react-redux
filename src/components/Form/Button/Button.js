import React from 'react';
import PropTypes from 'prop-types';
import { ButtonComponent } from './Button.styled';

const Button = ({onClick, title}) => (
  <ButtonComponent
    onClick={onClick}>
    {title}
  </ButtonComponent>
);

Button.propTypes = {
  onClick: PropTypes.func,
  title: PropTypes.string.isRequired
};

export default Button;
