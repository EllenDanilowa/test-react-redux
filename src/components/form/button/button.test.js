import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Button from './button';

describe('Button', () => {
  const title = 'Title';

  const createElement = (props, isSnapshot = true) => {
    const template = (<Button {...props} />);

    return isSnapshot ?
      renderer.create(template).toJSON() :
      shallow(template);
  };

  it('renders component with defined title and button type', () => {
    const type = 'button';
    const element = createElement({title, type});

    expect(element).toMatchSnapshot();
  });

  it('renders component with default props', () => {
    const element = createElement({title});

    expect(element).toMatchSnapshot();
  });

  it('sets click callback and triggers if clicked', () => {
    const onClick = jest.fn();
    const element = createElement({title, onClick}, false);

    element.simulate('click');

    expect(onClick).toHaveBeenCalled();
  });
});
