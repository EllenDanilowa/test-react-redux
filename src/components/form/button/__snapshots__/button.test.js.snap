// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders component with default props 1`] = `
<input
  className="sc-AxjAm gAbyhx"
  onClick={[Function]}
  type="button"
  value="Title"
/>
`;

exports[`Button renders component with defined title and button type 1`] = `
<input
  className="sc-AxjAm gAbyhx"
  onClick={[Function]}
  type="button"
  value="Title"
/>
`;
